apps/cowswap-frontend/src/legacy/hooks/useWrapCallback.ts

```diff
@@ -111,30 +138,40 @@ async function wrapContractCall(
   wethContract: Contract,
   amountHex: string,
   chainId: SupportedChainId,
-): Promise<TransactionResponse> {
+): Promise<WrapUnwrapTxData> {
   const estimatedGas = await wethContract.estimateGas.deposit({ value: amountHex }).catch(_handleGasEstimateError)
   const gasLimit = calculateGasMargin(estimatedGas)
 
   const network = await assertProviderNetwork(chainId, wethContract.provider, 'wrap')
 
   const tx = await wethContract.populateTransaction.deposit({ value: amountHex, gasLimit })
 
-  return wethContract.signer.sendTransaction({ ...tx, chainId: network })
+  const txResponse = await wethContract.signer.sendTransaction({ ...tx, chainId: network })
+
+  return {
+    txResponse,
+    tx,
+  }
```

Comment:
If this somehow fails or the bug is present but the user detects and decides to not place the tx, it won't be logged.
Maybe we should log instead before calling `sendTransaction`?

Reply:
It might be a problem, I can't say surely it won't.
Split the event into two:
 - `EthSendingIntention`
 - `EthSendingTransaction`

`EthSendingTransaction` will be send once we get txHash and it is connected to the sent intention event using `intentionEventId`.

Thanks!

https://github.com/cowprotocol/cowswap/pull/5380/commits/6dee1fef3b9e7fe32d8627b57db1ac8c7043fd3b